freqtrade/rpc/local_rest_server.py

```diff
@@ -0,0 +1,113 @@
+        :param freqtrade: Instance of a freqtrade bot
+        :return: None
+        """
+        super().__init__(freqtrade)
+        """ Constructor
+        :type interval: int
+        :param interval: Check interval, in seconds
+        """
+        self.interval = int(1)
+
+        thread = threading.Thread(target=self.run, args=(freqtrade,)) # extra comma as ref ! Tuple
+        thread.daemon = True                            # Daemonize thread
+        thread.start()     # Start the execution
+
+
+    def run(self, freqtrade):
+        """ Method that runs forever """
+        self._config = freqtrade.config
+        app = Flask(__name__)
+
+        """
+        Define the application routes here 
+        each Telegram command should have a like local substitute 
+        """
+        @app.route("/")
+        def hello():
+            # For simple rest server testing via browser
+            # cmds = 'Try uri:/daily?timescale=7 /profit /balance /status
+            #         /status /table /performance /count,
+            #         /start /stop /help'
+
+            rest_cmds ='Commands implemented: <br>' \
+                       '<a href=/daily?timescale=7>/daily?timescale=7</a>' \
+                       '<br>' \
+                       '<a href=/stop>/stop</a>' \
+                       '<br>' \
+                       '<a href=/start>/start</a>'
+            return rest_cmds
+
+        @app.route('/daily', methods=['GET'])
+        def daily():
+            try:
+                timescale = request.args.get('timescale')
+                logger.info("LocalRPC - Daily Command Called")
+                timescale = int(timescale)
+
+                stats = self._rpc_daily_profit(timescale,
+                                               self._config['stake_currency'],
+                                               self._config['fiat_display_currency']
+                                               )
+
+                stats = dumps(stats, indent=4, sort_keys=True, default=str)
+                return stats
+            except RPCException as e:
+                return e
+
+        @app.route('/start', methods=['GET'])
+        def start():
+            """
+            Handler for /start.
+            Starts TradeThread
+            """
+            msg = self._rpc_start()
+            print("msg is", msg)
+            return msg
+
+        @app.route('/stop', methods=['GET'])
+        def stop():
+            """
+            Handler for /stop.
+            Stops TradeThread
+            """
+            msg = self._rpc_stop()
+            print("msg is", msg)
+            return msg
+
+        """
+        Section to handle configuration and running of the Rest serve
+        also to check and warn if not bound to 127.0.0.1 as a security risk.
+        """
+
+        rest_ip = self._config['rest_cmd_line']['listen_ip_address']
+        rest_port = self._config['rest_cmd_line']['listen_port']
+
+        if rest_ip != "127.0.0.1":
```

Comment:
`127.0.0.1` is not the only local address. Please use IPv4Address for such things:
```
if not ipaddress.IPv4Address(rest_ip).is_loopback:
    ...
```

I think this information should be printed earlier somewhere in `freqtrade/configuration.py` like we do for other things and please don't spam log messages. One line should be enough :-)

Reply:
done